packages/office-ui-fabric-react/src/components/ActivityItem/ActivityItem.tsx

```diff
@@ -66,42 +66,59 @@ export class ActivityItem extends BaseComponent<IActivityItemProps, {}> {
   }
 
   @autobind
-  private _onRenderIcon(props: IActivityItemProps): JSX.Element {
+  private _onRenderIcon(props: IActivityItemProps): JSX.Element | null {
     if (props.activityPersonas) {
       return this._onRenderPersonaArray(props);
     }
+
+    return null;
   }
 
   @autobind
-  private _onRenderActivityDescription(props: IActivityItemProps): JSX.Element {
+  private _onRenderActivityDescription(props: IActivityItemProps): JSX.Element | null {
     if (props.activityDescriptionText) {
       return (<span className={ this._classNames.activityText }>{ props.activityDescriptionText }</span>);
     }
+
+    return null;
   }
 
   @autobind
-  private _onRenderComments(props: IActivityItemProps): JSX.Element {
+  private _onRenderComments(props: IActivityItemProps): JSX.Element | null {
     if (!props.isCompact && props.commentText) {
       return (<div className={ this._classNames.commentText }>{ props.commentText }</div>);
     }
+
+    return null;
   }
 
   @autobind
-  private _onRenderTimeStamp(props: IActivityItemProps): JSX.Element {
+  private _onRenderTimeStamp(props: IActivityItemProps): JSX.Element | null {
     if (!props.isCompact && props.timeStamp) {
       return (<div className={ this._classNames.timeStamp }>{ props.timeStamp }</div>);
     }
+
+    return null;
```

Comment:
Can't return undefined on something whose result is used directly in a render function